UOP1_Project/Assets/Scripts/Sound/SoundManager.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+
+public class SoundManager : MonoBehaviour
+{
+	public static SoundManager singleton;
+
+	[Header ("Init Variables")]
+	[Tooltip ("Amount of sound emitters created on Start")]
+	public int initialPoolSize = 1;
+
+	[Header ("Mixer Groups")]
+	public AudioMixerGroup masterMixer;
+	public AudioMixerGroup sfxMixer;
+	public AudioMixerGroup musicMixer;
+
+	private List<SoundEmitter> soundEmitterPool = new List<SoundEmitter>();
+	private GameObject soundEmittersContainer;
+	private int poolIndex = 0;
+
+	private void Awake()
+	{
+		DontDestroyOnLoad(this);
+
+		if (!SoundManager.singleton)
+			singleton = this;
+		else
+		{
+			Debug.LogWarning("One sound manager already present in the game, destroying " + this.name + ".");
+			Destroy(this);
+		}
+	}
+
+	void Start()
+    {
+		InitEmitterPool();
+    }
+
+	#region mixer groups functions
+	public static void ChangeVolumeOfMixerGroup(AudioMixerGroup mixer, float newVolumeNormalized)
+	{
+		mixer.audioMixer.SetFloat("Volume", NormalizedToMixerValue(newVolumeNormalized));
+	}
+
+	public static void SaveVolumeOfMixerGroup(AudioMixerGroup mixer)
+	{
+		float tempVolume;
+
+		if (mixer.audioMixer.GetFloat("Volume", out tempVolume))
+		{
+			PlayerPrefs.SetFloat(mixer.name, tempVolume);
+		}
+		else
+			Debug.LogError("Could not save volume for mixer group " + mixer.name + ". It does not contain an exposer variable with name Volume" );
+	}
+
+	public static void LoadVolumeOfMixerGroup(AudioMixerGroup mixer)
+	{
+		float tempVolume = PlayerPrefs.GetFloat(mixer.name, Mathf.Infinity);
+
+		if (tempVolume != Mathf.Infinity)
+		{
+			mixer.audioMixer.SetFloat("Volume", tempVolume);
+		}
+		else
+			Debug.Log("There is no saved volume preferences for mixer " + mixer.name + ", could not load volume.");
+	}
```

Comment:
I think setting/getting player prefs might be out of scope. Brief just calls for being able to set the volume via a settings menu. Probably best to leave changing player prefs to a settings menu or the save/load system.
```suggestion
	public static bool SetGroupVolume(AudioMixerGroup group, float volume)
	{
		return group.audioMixer.SetFloat("Volume", NormalizedToMixerValue(volume));
	}
	
	public static bool GetGroupVolume(AudioMixerGroup group, out float volume)
	{
	        return group.audioMixer.GetFloat("Volume", out volume);
	}
```